### ISSUE TITLE
Bloated XML in schemaXML of a taxonomy field

### ISSUE THREAD

**Issue by NPrice99** [OfficeDev/PnP#1110]
Hi If I Apply-SPOProvisioningTemplate and sometime later (The field in question has not been modified in between) Get-SPOProvisioningTemplate results in a lot of XML being returned for a Taxonomy field.  See below - 

<Field Type="TaxonomyFieldType" DisplayName="Project Structure" WebId="59eea740-3e15-49b8-942e-78f890024f6f" ShowField="Path1033" Required="FALSE" EnforceUniqueValues="FALSE" Group="XX Columns" ID="{f7f8b297-456e-47db-bb56-f54526d30a3b}" SourceID="{3622aa4b-4974-413e-9a7d-78b9904991d5}" StaticName="Project_x0020_Structure" Name="Project_x0020_Structure" List="Lists/TaxonomyHiddenList" AllowDeletion="TRUE">
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Default>
          </Default>
          <Customization>
            <ArrayOfProperty>
              <Property>
                <Name>SspId</Name>
                <Value xmlns:q1="http://www.w3.org/2001/XMLSchema" p4:type="q1:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">1203c05f-9d23-4eba-8d26-7b5e1fbd495e</Value>
              </Property>
              <Property>
                <Name>GroupId</Name>
              </Property>
              <Property>
                <Name>TermSetId</Name>
                <Value xmlns:q2="http://www.w3.org/2001/XMLSchema" p4:type="q2:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">33893f5f-e1ba-46f9-b340-7eb7278aaa81</Value>
              </Property>
              <Property>
                <Name>AnchorId</Name>
                <Value xmlns:q3="http://www.w3.org/2001/XMLSchema" p4:type="q3:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">00000000-0000-0000-0000-000000000000</Value>
              </Property>
              <Property>
                <Name>UserCreated</Name>
                <Value xmlns:q4="http://www.w3.org/2001/XMLSchema" p4:type="q4:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">false</Value>
              </Property>
              <Property>
                <Name>Open</Name>
                <Value xmlns:q5="http://www.w3.org/2001/XMLSchema" p4:type="q5:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">false</Value>
              </Property>
              <Property>
                <Name>TextField</Name>
                <Value xmlns:q6="http://www.w3.org/2001/XMLSchema" p4:type="q6:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">{2d52c241-8859-4475-859f-7165543580ff}</Value>
              </Property>
              <Property>
                <Name>IsPathRendered</Name>
                <Value xmlns:q7="http://www.w3.org/2001/XMLSchema" p4:type="q7:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">true</Value>
              </Property>
              <Property>
                <Name>IsKeyword</Name>
                <Value xmlns:q8="http://www.w3.org/2001/XMLSchema" p4:type="q8:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">false</Value>
              </Property>
              <Property>
                <Name>TargetTemplate</Name>
              </Property>
              <Property>
                <Name>CreateValuesInEditForm</Name>
                <Value xmlns:q9="http://www.w3.org/2001/XMLSchema" p4:type="q9:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">false</Value>
              </Property>
              <Property>
                <Name>FilterAssemblyStrongName</Name>
                <Value xmlns:q10="http://www.w3.org/2001/XMLSchema" p4:type="q10:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">Microsoft.SharePoint.Taxonomy, Version=15.0.0.0, Culture=neutral, PublicKeyToken=71e9bce111e9429c</Value>
              </Property>
              <Property>
                <Name>FilterClassName</Name>
                <Value xmlns:q11="http://www.w3.org/2001/XMLSchema" p4:type="q11:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">Microsoft.SharePoint.Taxonomy.TaxonomyField</Value>
              </Property>
              <Property>
                <Name>FilterMethodName</Name>
                <Value xmlns:q12="http://www.w3.org/2001/XMLSchema" p4:type="q12:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">GetFilteringHtml</Value>
              </Property>
              <Property>
                <Name>FilterJavascriptProperty</Name>
                <Value xmlns:q13="http://www.w3.org/2001/XMLSchema" p4:type="q13:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">FilteringJavascript</Value>
              </Property>
            </ArrayOfProperty>
          </Customization>
          <Default>
          </Default>
          <Customization>
            <ArrayOfProperty>
              <Property>
                <Name>SspId</Name>
                <Value xmlns:q1="http://www.w3.org/2001/XMLSchema" p4:type="q1:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">1203c05f-9d23-4eba-8d26-7b5e1fbd495e</Value>
              </Property>
              <Property>
                <Name>GroupId</Name>
              </Property>
              <Property>
                <Name>TermSetId</Name>
                <Value xmlns:q2="http://www.w3.org/2001/XMLSchema" p4:type="q2:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">33893f5f-e1ba-46f9-b340-7eb7278aaa81</Value>
              </Property>
              <Property>
                <Name>AnchorId</Name>
                <Value xmlns:q3="http://www.w3.org/2001/XMLSchema" p4:type="q3:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">00000000-0000-0000-0000-000000000000</Value>
              </Property>
              <Property>
                <Name>UserCreated</Name>
                <Value xmlns:q4="http://www.w3.org/2001/XMLSchema" p4:type="q4:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">false</Value>
              </Property>
              <Property>
                <Name>Open</Name>
                <Value xmlns:q5="http://www.w3.org/2001/XMLSchema" p4:type="q5:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">false</Value>
              </Property>
              <Property>
                <Name>TextField</Name>
                <Value xmlns:q6="http://www.w3.org/2001/XMLSchema" p4:type="q6:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">{2d52c241-8859-4475-859f-7165543580ff}</Value>
              </Property>
              <Property>
                <Name>IsPathRendered</Name>
                <Value xmlns:q7="http://www.w3.org/2001/XMLSchema" p4:type="q7:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">true</Value>
              </Property>
              <Property>
                <Name>IsKeyword</Name>
                <Value xmlns:q8="http://www.w3.org/2001/XMLSchema" p4:type="q8:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">false</Value>
              </Property>
              <Property>
                <Name>TargetTemplate</Name>
              </Property>
              <Property>
                <Name>CreateValuesInEditForm</Name>
                <Value xmlns:q9="http://www.w3.org/2001/XMLSchema" p4:type="q9:boolean" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">false</Value>
              </Property>
              <Property>
                <Name>FilterAssemblyStrongName</Name>
                <Value xmlns:q10="http://www.w3.org/2001/XMLSchema" p4:type="q10:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">Microsoft.SharePoint.Taxonomy, Version=15.0.0.0, Culture=neutral, PublicKeyToken=71e9bce111e9429c</Value>
              </Property>
              <Property>
                <Name>FilterClassName</Name>
                <Value xmlns:q11="http://www.w3.org/2001/XMLSchema" p4:type="q11:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">Microsoft.SharePoint.Taxonomy.TaxonomyField</Value>
              </Property>
              <Property>
                <Name>FilterMethodName</Name>
                <Value xmlns:q12="http://www.w3.org/2001/XMLSchema" p4:type="q12:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">GetFilteringHtml</Value>
              </Property>
              <Property>
                <Name>FilterJavascriptProperty</Name>
                <Value xmlns:q13="http://www.w3.org/2001/XMLSchema" p4:type="q13:string" xmlns:p4="http://www.w3.org/2001/XMLSchema-instance">FilteringJavascript</Value>
              </Property>
            </ArrayOfProperty>
          </Customization>

Using SharePoint Manager 2013 the SchemaXML is the same as above.

Text, Note, MultiUSer, HTML, Lookup, Choice, User fields all OK

**Comment by NPrice99** [comment]
Hi Irwin
I ran a controlled experiment tonight.
Brand new Farm nothing in it other than the SharePoint Binaries and having AutoSPInstaller run on it.

No Site Fields, No Content Types. Created a Web Application, Created a Site Collection using Template #STS. (All On Premise)

Ran Apply-SPOProvisioningTemplate.

Had 7 failed runs which resulted in rerunning Apply-SPOProvisioningTemplate.

3 failures were Duplicate Fields see #1116 - I removed the Note Field causing the issue.
3 Were Missing features and In just removed the features from the ProvisioningTemplate
1 Was "Value does not fall within Range"  Caused by Lookup fields not having the Lookup List in the SchemaXml.  An old bug I raised in June or July.  I have a PowerShell script which repairs these Lookup Fields.

This resulted in the same field as above (Project_x0020_Structure) Having 4 copies of <Default></Default> and 4 copies of <Customisation>...........</Customisation> in the Field SchemaXml.  I can confirm that there was only one copy of each of these in the Provisioning Template.
